Added Dawn profile for Gnome Terminal

diorcula opened the pull request:
- still needs testing as this is my first ever contribution to a gnome profile theme.

diorcula committed:
:sparkles: :lipstick: added dawn colour profile for Gnome

diorcula committed:
:coffin: removed unused not related themes from dconf

mvllow commented on readme.md:
```diff
@@ -1,6 +1,6 @@
 <p align="center">
     <img src="https://github.com/rose-pine/rose-pine-theme/raw/main/assets/icon.png" width="80" />
-    <h2 align="center">Rosé Pine for GNOME Terminal</h2>
+    <h2 align="center">Rosé Pine Dawn for GNOME Terminal</h2>
```
The dawn variant should be in addition, not instead of the base variant

diorcula replied:
sorry i forgot about this! i fixed it and will make a new PR!

mvllow reviewed (changes requested):
Thank you for taking the time to contribute this. Please add Rosé Pine dawn as a separate file with only relevant parts of that theme. Our base theme does have some unnecessary bits which you pointed out but that should be another PR 😌

If it's any easier, feel free to revise the PR to only add `rosepine-dawn.dconf` and we can update the readme afterwards

diorcula committed:
:ambulance: added the pre-existing theme back in

in accordance to the comment on the PR; also removed 'Regolith' theme from the files as they deemed unnecessary

mvllow commented:
Thank you!
